### PR TITLE
Routes created and added style to navigation

### PR DESCRIPTION
**Pull request by linhapeter**

### CODE REVIEW

File: src/style/header.css
Comment (volkov-a-dev) [inline src/style/header.css:1]: Why not connect scss?

File: src/style/chat.css
Comment (volkov-a-dev) [inline src/style/chat.css:1]: Why not connect scss?

File: src/pages/home/Home.jsx
Comment (volkov-a-dev) [inline src/pages/home/Home.jsx:1]: If the component should return only JSX, without any manipulations with props, store, or local state. You can shorten this code to:
```
import React from "react";
const Home = () => (
    <div>
        <h1>Home</h1>
    </div>
);

export default Home;
```

File: src/pages/examples/form/ResponseContent.jsx [hunk 1]
```diff
@@ -13,4 +14,13 @@ const ResponseContent = ({ response }) => (
   </div>
 );
 
+ResponseContent.propTypes = {
```
Comment (volkov-a-dev) [inline src/pages/examples/form/ResponseContent.jsx]: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/require-default-props.md
propType " response" is not required, but has no corresponding defaultProps declaration.

ResponseContent.propTypes = {
  response: PropTypes.shape({
    json: PropTypes.shape({
      name: PropTypes.string,
      password: PropTypes.string,
    }),
  }),
};

ResponseContent.defaultProps = {
  response: {
     json: {
       name: '',
       password: ''
     }
  }
}

File: src/header/Header.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          </li>
+          <li className="nav-links__li">
+            <Link className="nav-links__a" to="/login">
+              Login
+            </Link>
+          </li>
+        </ul>
+      </nav>
+
+      <Routes>
```
Comment (volkov-a-dev) [inline src/header/Header.jsx]: https://reactrouter.com/en/main/router-components/browser-router
You need to move the routing to a separate folder and make it more isolated. Also, why not write a loop that would go through the array and return the same thing as here?
Advice: If it is possible to automate a process, automate it

File: src/header/Header.jsx
Comment (volkov-a-dev) [inline src/header/Header.jsx:1]: It's better to create a folder for the **Сomponents** and move the component headers there.
This folder will store those components that will be used on different pages.

**Review by volkov-a-dev — CHANGES_REQUESTED** [review]

File: src/index.js [hunk 3]
```diff
@@ -1,6 +1,11 @@
 import React from "react";
 import { createRoot } from "react-dom/client";
 import App from "./App.jsx";
+import { BrowserRouter, BrowserRouter as Router } from "react-router-dom";
```
Comment (volkov-a-dev) [inline src/index.js]: 'Router' is declared but its value is never read.
'Router' is defined but never used.eslint[no-unused-vars](https://eslint.org/docs/latest/rules/no-unused-vars)

File: src/AppRouting.jsx
Comment (volkov-a-dev) [inline src/AppRouting.jsx:1]: I'd rather put this in a separate folder: routing

File: src/components/header/Header.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import "../../style/header.scss";
+import AppRouting from "../../AppRouting";
+
+const routes = [
```
Comment (volkov-a-dev) [inline src/components/header/Header.jsx]: I would put this in constants or better yet in a config file. Where there were two objects, one contained the route constants. The second contained router data. Below is an example. I would also store everything in the routes folder
**routeConfig.js**
```
export const ROUTES_LIST = [
  {
          id: 1,
          name: 'home page',
          value: 'Home Page',
          path: '/',
  },
  {
          id: 2,
          name:  'examples',
          value: 'Examples',
          path: '/examples',
  }
   .....
```
```
export const routes = [
    {
        path:  ROUTES_LIST.path,
        component: HomePage,
        exact: true
    },
    {
        path: ROUTES_LIST.examples,
        component: Examples,
        exact: true
    },
    ....
]

```

File: src/components/header/Header.jsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  <li className="nav-links__li" key={id}>
+    <Link className="nav-links__a" to={route.path}>
+      {route.label}
+    </Link>
+  </li>
+));
+
+const Header = () => {
+  return (
+    <div>
```
Comment (volkov-a-dev) [inline src/components/header/Header.jsx]: Why is there a div here?

File: src/pages/examples/messenger/Chat.jsx [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import React, { useState } from "react";
 import { v4 as uuidv4 } from "uuid";
 import Message from "./Message";
+import "../../../style/chat.scss";
```
Comment (volkov-a-dev) [inline src/pages/examples/messenger/Chat.jsx]: Everything that touches a component must be in the component's component. chat

**Review by volkov-a-dev — CHANGES_REQUESTED** [review]

**Comment by linhapeter** [comment]
I have tried apply this logic in Examples.jsx  but EXAMPLES_ROUTES_LIST was always empty. Any idea what went wrong?

import { EXAMPLES_ROUTES_LIST } from "../../routing/config"

const renderedLinks = EXAMPLES_ROUTES_LIST?.map((route, id) => (
  <li key={id}>
    <Link to={route.path}>
      {route.value}
    </Link>
  </li>
));

const Examples = () => (
  <div>
    <ul>
      {renderedLinks}
    </ul>
    <Outlet />
  </div>
);

export default Examples;

**Pull request merged**